brvs-core/src/main/java/iroha/validation/transactions/core/provider/impl/RegisteredUsersStorageImpl.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package iroha.validation.transactions.core.provider.impl;
+
+import static com.mongodb.client.model.Filters.eq;
+
+import com.mongodb.client.model.ReplaceOptions;
+import iroha.validation.transactions.core.archetype.mongo.MongoBasedStorage;
+import iroha.validation.transactions.core.provider.RegisteredUsersStorage;
+import iroha.validation.transactions.core.provider.impl.RegisteredUsersStorageImpl.UserAccountId;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class RegisteredUsersStorageImpl extends MongoBasedStorage<UserAccountId>
+    implements RegisteredUsersStorage {
+
+  private static final ReplaceOptions optionsToKeep = new ReplaceOptions().upsert(false);
+  private static final String USER_NAME_ATTRIBUTE = "userId";
+  private static final int DEFAULT_PAGE_SIZE = 500;
+  private static final String DEFAULT_DB_NAME = "userStorage";
+  private static final String DEFAULT_COLLECTION_NAME = "users";
+
+  public RegisteredUsersStorageImpl(String mongoHost, int mongoPort) {
+    super(mongoHost, mongoPort, DEFAULT_DB_NAME, DEFAULT_COLLECTION_NAME, UserAccountId.class);
+  }
+
+  @Override
+  public void add(String accountId) {
+    collection.replaceOne(eq(USER_NAME_ATTRIBUTE, accountId),
+        new UserAccountId(accountId),
+        optionsToKeep
+    );
+  }
+
+  @Override
+  public boolean contains(String accountId) {
+    return collection.find(eq(USER_NAME_ATTRIBUTE, accountId)).first() == null;
+  }
+
+  @Override
+  public <T> Set<T> process(Function<Iterable<String>, Collection<T>> method) {
+    final Set<T> resultSet = new HashSet<>();
+    int pageCounter = 1;
```

Comment:
I can not see any place where pageCounter is changed. Look's like it is always equal to 1

Reply:
By the way, you can set initial value of `pageCounter` to 0 and remove decrementing in `getAccountsPage()`

Reply:
Thanks for the observation!